crowbar_framework/app/models/cinder_service.rb

```diff
@@ -68,11 +68,11 @@ def create_proposal
     base = super
 
     nodes = NodeObject.all
-    controller = select_nodes_for_role(nodes, "cinder-controller", "controller") || []
+    controllers = select_nodes_for_role(nodes, "cinder-controller", "controller") || []
     storage = select_nodes_for_role(nodes, "cinder-volume", "storage") || []
 
     base["deployment"][@bc_name]["elements"] = {
-      "cinder-controller" => [ controller.first.name ],
+      "cinder-controller" => controllers.empty? [] : [ controllers.first.name ],
```

Comment:
> or perhaps use the map construct, since controller should never return more than one node.. (since it evaluates constraints, and that says unique.. )

It actually doesn't evaluate the unique constraint yet. That's a good item to fix, though :-)